arch/sim/include/setjmp.h

```diff
@@ -111,7 +145,8 @@
 
 #ifndef __ASSEMBLY__
 
-#if defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)
+#if (defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)) || \
+     defined(CONFIG_HOST_ARM64)
```

Comment:
typedef unsigned long xcpt_reg_t should work in all case

Reply:
Maybe, but "#if defined(CONFIG_HOST_X86_64) && !defined(CONFIG_SIM_M32)" was already in the code. I'm not sure, if other simulations will break. I get remove it and let the CI check?

Reply:
I removed it.

Reply:
"ypedef unsigned long xcpt_reg_t" is not enough. Linux(sim-02) breaks:

    In file included from sim/up_vfork.c:32:
    sim/up_vfork.c: In function 'up_vfork':
    Error: sim/up_vfork.c:96:9: error: format '%x' expects argument of type 'unsigned int', but argument 3 has type 'xcpt_reg_t' {aka 'const long unsigned int'} [-Werror=format=]
     96 |   sinfo("  frame pointer:%08" PRIxPTR " sp:%08" PRIxPTR " pc:%08" PRIxPTR ""
        |         ^~~~~~~~~~~~~~~~~~~~~
     97 |         "\n", context[JB_FP], context[JB_SP], context[JB_PC]);
        |               ~~~~~~~~~~~~~~
        |                      |
        |                      xcpt_reg_t {aka const long unsigned int}
    In file included from /github/workspace/sources/nuttx/include/stdint.h:35,
                 from /github/workspace/sources/nuttx/include/sys/types.h:32,
                 from /github/workspace/sources/nuttx/include/stddef.h:28,
                 from /github/workspace/sources/nuttx/include/inttypes.h:28,
                 from sim/up_vfork.c:27:
    /github/workspace/sources/nuttx/include/arch/inttypes.h:125:24: note: format string is defined here
    125 | #  define PRIxPTR     "x"

Reply:
The code need adjust too, please see #5473

Reply:
Okay ... let's wait until that PR is integrated.